storage/postgres/storage.go

```diff
@@ -245,6 +246,13 @@ func (ps *Storage) Delete(ctx context.Context, objType types.ObjectType, criteri
 	rows, err := ps.queryBuilder.NewQuery(entity).WithCriteria(criteria...).Return("*").Delete(ctx)
 	defer closeRows(ctx, rows)
 	if err != nil {
+		pqError, ok := err.(*pq.Error)
+		if ok && pqError.Code.Name() == "foreign_key_violation" {
```

Comment:
Extract constant "foreign_key_violation" or I think there are predefined code names in pq package

Reply:
We make use of the _ foreign_key_violation_ string, as this is what pq returns: https://github.com/lib/pq/blob/f91d3411e481ed313eeab65ebfe9076466c39d01/error.go#L177

Unfortunately, beyond a plain a string value to compare with, they don't have an explicit error type which we can assert, to ensure for such an error.